File: Python/129_Sum_Root_to_leaf_ numbers.py
class Solution:
    def sumNumbers(self, root: TreeNode) -> int:
        return self.helper(0, root)

    def helper(self, num, root):
        if not root:
            return num
        left = self.helper(num * 10 + root.val, root.left)
        right = self.helper(num * 10 + root.val, root.right)
        if not root.right:
            return left
        elif not root.left:
            return right
        return left + right

File: leetcode/archived/golang/129_Sum_Root_To_Leaf_Numbers.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumNumbers(root *TreeNode) int {
	return helper(root, 0 )
}

func helper(root *TreeNode, sum int) int{
	if root == nil{
		return sum
	}
	left := helper(root.Left, sum*10+root.Val)
	right := helper(root.Right, sum*10+root.Val)
	if root.Left ==  nil{
		return right
	}
	if root.Right == nil{
		return left
	}
	return left+right
}